blocks/link-editor/index.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * External dependencies
+ */
+import { isBoolean, noop } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
+import { IconButton } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import UrlInput from '../url-input';
+import { filterURLForDisplay } from '../../editor/utils/url';
+import ToggleControl from '../inspector-controls/toggle-control';
+
+export default class LinkEditor extends Component {
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			settingsVisible: false,
+			newLinkValue: '',
+			isEditing: null,
+		};
+
+		this.onChangeLinkValue = this.onChangeLinkValue.bind( this );
+		this.toggleLinkSettingsVisibility = this.toggleLinkSettingsVisibility.bind( this );
+		this.submitLink = this.submitLink.bind( this );
+		this.editClicked = this.editClicked.bind( this );
+		this.dropClicked = this.dropClicked.bind( this );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( isBoolean( nextProps.isEditing ) ) {
+			this.setState( { isEditing: nextProps.isEditing } );
+		} else if ( this.state.isEditing === null ) {
+			this.setState( { isEditing: ! nextProps.linkValue } );
```

Comment:
What's the reasoning here? I don't understand why we have to set `isEditing` to true when the link value changes?

Reply:
The reasoning here is if we receive and edition prop we just use it.
If not, if the edition state is already defined we continue to use it and do nothing.
If the edition state is not defined we check if a link value exists if yes we start without edition e.g a link already exists the user can see it but needs to press edit to edit.
If a link does not exist we start with edition, it means we will insert a new link and we can not preview and click on a link yet.